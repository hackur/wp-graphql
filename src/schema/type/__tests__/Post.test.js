import PostType from 'type/Post';

// eslint-disable-next-line no-underscore-dangle
const fields = PostType._typeConfig.fields();

describe('Test Post type', () => {
  test('Test name', () => {
    expect(PostType.name).toMatchSnapshot();
  });

  test('Test type', () => {
    expect(PostType.isTypeOf({ type: 'post' })).toBe(true);
  });

  test('Test fields', () => {
    expect(Object.keys(PostType.getFields())).toMatchSnapshot();
  });

  test('Test resolve categories', async () => {
    const categories = await fields.categories.resolve({ categories: [69] });
    expect(categories).toMatchSnapshot();
  });

  test('Test resolve categories null', () => {
    expect(fields.categories.resolve({ categories: [] })).toBeNull();
  });

  test('Test resolve tags', async () => {
    const tags = await fields.tags.resolve({ tags: [69] });
    expect(tags).toMatchSnapshot();
  });

  test('Test resolve tags null', () => {
    expect(fields.tags.resolve({ tags: [] })).toBeNull();
  });
});
