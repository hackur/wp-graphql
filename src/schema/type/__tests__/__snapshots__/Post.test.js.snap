// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Post type Test fields 1`] = `
Array [
  "id",
  "date",
  "date_gmt",
  "guid",
  "modified",
  "modified_gmt",
  "slug",
  "type",
  "link",
  "title",
  "content",
  "excerpt",
  "comment_status",
  "ping_status",
  "template",
  "format",
  "author",
  "featuredMedia",
  "meta",
  "sticky",
  "categories",
  "tags",
  "comments",
]
`;

exports[`Test Post type Test name 1`] = `"Post"`;

exports[`Test Post type Test resolve categories 1`] = `
Array [
  Category {
    "id": 69,
  },
]
`;

exports[`Test Post type Test resolve tags 1`] = `
Array [
  Tag {
    "id": 69,
  },
]
`;
